cipher/xor/xor.go

```diff
@@ -2,24 +2,58 @@
 // ref: https://en.wikipedia.org/wiki/XOR_cipher
 package xor
 
+import "unsafe"
```

Comment:
I don't understand why you have to use unsafe here?

Reply:
The reason for using unsafe is to convert []byte to []uint64, byte xor byte is far less efficient than uin64 xor uint64.

Reply:
I see that there is an open issue in Golang about this. I don't know why this is the case as byte is a type alias for uint8.